src/Microsoft.Data.Analysis/Converters.cs

```diff
@@ -0,0 +1,948 @@
+                return (IFloatConverter<T>)new UShortFloatConverter();
+            }
+            if (typeof(T) == typeof(int))
+            {
+                return (IFloatConverter<T>)new IntFloatConverter();
+            }
+            if (typeof(T) == typeof(uint))
+            {
+                return (IFloatConverter<T>)new UIntFloatConverter();
+            }
+            if (typeof(T) == typeof(long))
+            {
+                return (IFloatConverter<T>)new LongFloatConverter();
+            }
+            if (typeof(T) == typeof(ulong))
+            {
+                return (IFloatConverter<T>)new ULongFloatConverter();
+            }
+            if (typeof(T) == typeof(float))
+            {
+                return (IFloatConverter<T>)new FloatFloatConverter();
+            }
+            throw new NotSupportedException();
+        }
+    }
+    internal class ByteFloatConverter : IFloatConverter<byte>
+    {
+        public float GetFloat(byte value)
+        {
+            return (float)value;
+        }
+    }
+    internal class SByteFloatConverter : IFloatConverter<sbyte>
+    {
+        public float GetFloat(sbyte value)
+        {
+            return (float)value;
+        }
+    }
+    internal class ShortFloatConverter : IFloatConverter<short>
+    {
+        public float GetFloat(short value)
+        {
+            return (float)value;
+        }
+    }
+    internal class UShortFloatConverter : IFloatConverter<ushort>
+    {
+        public float GetFloat(ushort value)
+        {
+            return (float)value;
+        }
+    }
+    internal class IntFloatConverter : IFloatConverter<int>
+    {
+        public float GetFloat(int value)
+        {
+            return (float)value;
+        }
+    }
+    internal class UIntFloatConverter : IFloatConverter<uint>
+    {
+        public float GetFloat(uint value)
+        {
+            return (float)value;
+        }
+    }
+    internal class LongFloatConverter : IFloatConverter<long>
+    {
+        public float GetFloat(long value)
+        {
+            return (float)value;
+        }
+    }
+    internal class ULongFloatConverter : IFloatConverter<ulong>
+    {
+        public float GetFloat(ulong value)
+        {
+            return (float)value;
+        }
+    }
+    internal class FloatFloatConverter : IFloatConverter<float>
+    {
+        public float GetFloat(float value)
+        {
+            return (float)value;
+        }
+    }
+    internal interface IDoubleConverter<T>
+    {
+        double GetDouble(T value);
+    }
+    internal static class DoubleConverter<T>
+    {
+        public static IDoubleConverter<T> Instance { get; } = DoubleConverter.GetDoubleConverter<T>();
+    }
+    internal static class DoubleConverter
+    {
+        public static IDoubleConverter<T> GetDoubleConverter<T>()
+        {
+            if (typeof(T) == typeof(byte))
+            {
+                return (IDoubleConverter<T>)new ByteDoubleConverter();
```

Comment:
(nit) is the cast necessary here?

Reply:
Yup, an implicit conversion does not exist between IDoubleConverter<T> and ByteDoubleConverter (even though it implements IDoubleConverter<byte>)